### DISCUSSION TITLE
20231019 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#62] [Meetings]
Date: 2023-10-19

Time: 8:30 UTC (10:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - 
- CBosse(@cgbosse) - 
- MMelchor(@mario-nt) - 
- GByrne(@grmbyrn) - 

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by grmbyrn** [comment]
## Python
* Took course on Python, learning from Python Operators to Python Virtual Environments and pip.
* Completed Rock Paper Scissors game in Python, which can be found [here](https://github.com/grmbyrn/rock_paper_scissors).
* Wrote article explaining Python Rock Paper Scissors, which can be found [here](https://medium.com/@grmbrn89/rock-paper-scissors-game-in-python-explained-94be0edbcb68)

## AI
* Continued course on using OpenAI, which includes [this](https://github.com/grmbyrn/know-it-all-ai) project on how to create a chatbot AI.

**Comment by mario-nt** [comment]
### GUI

- Worked on https://github.com/torrust/torrust-index-gui/issues/277
- Worked on https://github.com/torrust/torrust-index-gui/issues/229
- Worked on https://github.com/torrust/torrust-index-gui/issues/232
- Wordked on https://github.com/torrust/torrust-index-gui/issues/310
### INDEX
- Worked on https://github.com/torrust/torrust-index/issues/39
### Other

- Closed https://github.com/torrust/torrust-index-api-lib/issues/37
### Training
**Rust** 

- Traits
- state pattern
- Smart pointer Box dyn
- Trait bound | Where clause
- Structs and instantiation
- freeCodeAcademy Rust Introduction Course
 
**Vue**

- 	Components basics
- 	Props
- Auto imports

**Typescript**

- Types, supertype and subtype

**Cypress**

- Cypress 101 (official docs and freeCodeCamp course)
- Cypress's best practices

**Comment by josecelano** [comment]
I've been sick with a neck pain :-( since the 9th so I can concentrate very weel I'm been stretching and going to the physiotherapist a couple of times.

### Torrust

- I caught up with @da2ce7's work
- I review @da2ce7's PR about labels: https://github.com/torrust/torrust-tracker/pull/482#pullrequestreview-1680402504
- I was discussing with @mario-nt about E2E testing in the Index Frontend GUI, especially how we could/should clean the app state after running a test.

#### Torrust Index

#### Torrusdt Tracker

- Merged PR from @si14 https://github.com/torrust/torrust-tracker/pull/477#issuecomment-1768321509 to use his package to identify peer ids. We want to increase collaboration with other things like the info-hash code. Do you thing @a2ce7 we should use a common GitHub org? For the time being, I think we can continue using the packages where there are.

- Reviweing PR to add a workflow to release on Github: https://github.com/torrust/torrust-tracker/pull/473

### Serde-bencode

Now I'm a maintainer and I can publish new versions. I've publish a new version 0.2.4 to fix the problem we had here: https://github.com/torrust/torrust-index/issues/266.

There are some issues pending to fix: https://github.com/toby/serde-bencode/issues @da2ce7 should we spend more time fixing them? Although it's a small package, It's not simple. I was watching this video https://youtu.be/BI_bHCGRgMY to understand how serde works. And I think kIt will take me some days to fix all the open issues. 

I do not like the way tests are organized. I would like to organize them in a different way:

- Unit tests for building generic Rsut struct
- Units test for serializing and deserializing with all cases ordered
- Regression tests

### Secure Git Guide

- I have a meeting a¡with @grmbyrn there is a new PR pending to review and merge: https://github.com/nautilus-cyberneering/secure-git-guide/pull/69. But he has to fix megalinter errors.

### Learning:

- https://www.youtube.com/watch?v=bG1aq7eUyiE (Connecting BitTorrent to the Blockchain - Keynote with Justin Knoll at niTROn 2019)
- https://www.youtube.com/watch?v=97qyNQz7fxY (Continuous Integration: That’s Not What They Meant • Clare Sudbery • YOW! 2023)
- https://open.spotify.com/episode/3Dn04IuwiCdbONcWKWBcIZ?si=ykyC-UD5SjyvtQZTCrp2YA (Postcast about Product Teams - Spanish)
- https://youtu.be/BI_bHCGRgMY (Decrusting the serde crate)
- Finish reading book_ Sustainable Code - Spanish (https://savvily.es/libros/codigo-sostenible/)

**Comment by josecelano** [comment]
Hi @mario-nt @da2ce7 @grmbyrn @WarmBeer, I've opened a discussion https://github.com/torrust/torrust-index/discussions/356 about what we were talking about in the meeting.
